ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/PluginStatusDialog.java

```diff
@@ -115,8 +120,9 @@ public void showLink(boolean showLink) {
 		fShowLink = showLink;
 	}
 
-	public void setInput(Map<?, ?> input) {
+	public void setInput(Map<?, ?> input, ILaunchConfiguration iLaunchConfiguration) {
 		fInput = input;
+		fLaunchConfiguration = iLaunchConfiguration;
```

Comment:
Here you could just pass the `LaunchValidationOperation` and obtain the input and launch-config just within this method. This would simplify all callers to just pass the operation instead repeating the unwrapping again and again.
```suggestion
	public void setInput(LaunchValidationOperation operation) {
		fInput = operation.getInput();
		fLaunchConfiguration = operation.getLaunchConfiguration();
```